hw/mcu/dialog/da1469x/src/da1469x_dma.c

```diff
@@ -116,17 +122,19 @@ da1469x_dma_init(void)
 struct da1469x_dma_regs *
 da1469x_dma_acquire_single(int cidx)
 {
-    struct da1469x_dma_regs *chan;
+    struct da1469x_dma_regs *chan = NULL;
+    int sr;
 
     assert(cidx < MCU_DMA_CHAN_MAX);
 
+    OS_ENTER_CRITICAL(sr);
     if (cidx < 0) {
         cidx = find_free_single();
         if (cidx < 0) {
-            return NULL;
+            goto end_single;
```

Comment:
@nkaje I think you want to still return NULL for the two error cases, `if (cidx < 0)` and `else if (g_da1469x_dma_acquired & (1 << cidx)`

Reply:
`chan` is initialized to NULL, which gets set by `MCU_DMA_CIDX2CHAN(cidx)` down below. If we end up taking the `goto` path to `end_single`, `chan` set to `NULL` is returned.

Reply:
@nkaje Ahh.. I missed it. Thanks, it looks good.